src/shells/layer_shell.rs

```diff
@@ -0,0 +1,141 @@
+    reexports::{
+        client::{
+            Connection, Dispatch, Proxy, QueueHandle, globals::GlobalList,
+            protocol::wl_surface::WlSurface,
+        },
+        protocols_wlr::layer_shell::v1::client::{zwlr_layer_shell_v1, zwlr_layer_surface_v1},
+    },
+    shell::{
+        WaylandSurface,
+        wlr_layer::{LayerShell, LayerShellHandler, LayerSurface, LayerSurfaceData},
+    },
+};
+
+pub use smithay_client_toolkit::shell::wlr_layer::{Anchor, KeyboardInteractivity, Layer};
+
+use crate::state::SmithayRunnerState;
+
+#[derive(Debug, Resource, Component, Clone)]
+pub struct LayerShellSettings {
+    pub anchor: Anchor,
+    pub size: (u32, u32),
+    pub exclusive_zone: i32,
+    pub margin: (i32, i32, i32, i32),
+    pub keyboard_interactivity: KeyboardInteractivity,
+    pub layer: Layer,
+}
+
+impl Default for LayerShellSettings {
+    fn default() -> Self {
+        Self {
+            anchor: Anchor::empty(),
+            exclusive_zone: Default::default(),
+            margin: Default::default(),
+            size: (256, 256),
+            keyboard_interactivity: Default::default(),
+            layer: Layer::Top,
+        }
+    }
+}
+
+pub struct LayerShellWindow {
+    window: LayerSurface,
+    conn: Connection,
+}
+
+impl LayerShellWindow {
+    pub fn layer_surface(&self) -> &LayerSurface {
+        &self.window
+    }
+
+    pub fn layer_surface_mut(&mut self) -> &mut LayerSurface {
+        &mut self.window
+    }
+}
+
+impl HasWindowHandle for LayerShellWindow {
+    fn window_handle(
+        &self,
+    ) -> Result<raw_window_handle::WindowHandle<'_>, raw_window_handle::HandleError> {
+        let raw_window_handle = RawWindowHandle::Wayland(WaylandWindowHandle::new(
+            core::ptr::NonNull::new(self.window.wl_surface().id().as_ptr() as *mut _).unwrap(),
+        ));
+        unsafe { Ok(WindowHandle::borrow_raw(raw_window_handle)) }
+    }
+}
+
+impl HasDisplayHandle for LayerShellWindow {
+    fn display_handle(
+        &self,
+    ) -> Result<raw_window_handle::DisplayHandle, raw_window_handle::HandleError> {
+        let raw_display_handle = RawDisplayHandle::Wayland(WaylandDisplayHandle::new(
+            core::ptr::NonNull::new(self.conn.backend().display_ptr() as *mut _).unwrap(),
+        ));
+        unsafe { Ok(DisplayHandle::borrow_raw(raw_display_handle)) }
+    }
+}
+
+pub fn create_window<State>(
+    globals: &GlobalList,
+    qh: &QueueHandle<State>,
+    surface: WlSurface,
+    conn: Connection,
+    settings: &LayerShellSettings,
+) -> LayerShellWindow
+where
+    State: LayerShellHandler
+        + Dispatch<zwlr_layer_shell_v1::ZwlrLayerShellV1, GlobalData>
+        + Dispatch<zwlr_layer_surface_v1::ZwlrLayerSurfaceV1, LayerSurfaceData>
+        + 'static,
+{
+    let layer_shell = LayerShell::bind(globals, qh).expect("layer shell not available");
+    let layer =
+        layer_shell.create_layer_surface(qh, surface, settings.layer, Some("simple_layer"), None);
```

Comment:
You need to expose this "simple_layer" this might refer to app_id which is important for certain wayland functions. You can set default to `layer_shell_app` if user has not passed `app_id`